Sources/Build/BuildDescription/ProductBuildDescription.swift

```diff
@@ -221,7 +235,7 @@ public final class ProductBuildDescription: SPMBuildCore.ProductBuildDescription
             // version of the package that defines the executable product.
             let executableTarget = try product.executableTarget
             if executableTarget.underlyingTarget is SwiftTarget, self.toolsVersion >= .v5_5,
-               self.buildParameters.canRenameEntrypointFunctionName
+               self.buildParameters.canRenameEntrypointFunctionName, executableTarget.underlyingTarget.type != .macro
```

Comment:
this will mean we cannot test these as we do executables, is that desired?

Reply:
It is right now, but should get fixed for the final PR or in a follow-up. Since it needs to take `BUILD_MACROS_AS_DYLIBS` into account, I wanted to avoid some complexity here while developing.

Reply:
Going to leave this as-is for this PR and resolve once compiler support for macros as executables is present.